sui/gateway.js

```diff
@@ -510,6 +509,65 @@ async function testNewField(value, options) {
     console.log(`Set the value to ${value} and it was set to ${returnedValue}.`);
 }
 
+async function pause(keypair, client, config, chain, contracts, args, options) {
+    const response = await client.getObject({
+        id: contracts.objects.Gatewayv0,
+        options: {
+            showContent: true,
+            showBcs: true,
+        },
+    });
+    let allowedFunctionsArray = response.data.content.fields.value.fields.version_control.fields.allowed_functions;
+    allowedFunctionsArray = allowedFunctionsArray.map((allowedFunctions) => allowedFunctions.fields.contents);
+
+    const versionsArg = [];
+    const allowedFunctionsArg = [];
+
+    for (const version in allowedFunctionsArray) {
+        const allowedFunctions = allowedFunctionsArray[version];
+
+        // Do not dissalow `allow_function` because that locks the gateway forever.
+        if (Number(version) === allowedFunctionsArray.length - 1) {
+            const index = allowedFunctions.indexOf('allow_function');
+
+            if (index > -1) {
+                // only splice array when item is found
+                allowedFunctions.splice(index, 1); // 2nd parameter means remove one item only
+            }
+        }
+
+        printInfo(`Functions that will be disallowed for version ${version}`, allowedFunctions);
+
+        versionsArg.push(new Array(allowedFunctions.length).fill(version).join());
+        allowedFunctionsArg.push(allowedFunctions.join());
+    }
+
+    // Write the
+    writeJSON(
+        {
+            versions: versionsArg,
+            disallowedFunctions: allowedFunctionsArg,
+        },
+        `${__dirname}/../axelar-chains-config/info/sui-gateway-allowed-functions-${options.env}.json`,
```

Comment:
save to the existing `sui.contracts.AxelarGateway` config which is already included in git

Reply:
Instead of dynamically querying this, we should look into creating a json file in the cgp sui that lists the version controlled methods, and including it in the npm release. We already do something like this for the golden test, so shouldn't be hard to add a JSON file containing contract name -> allowed function list mapping